doc/specs/Keyboard-Selection.md

```diff
@@ -0,0 +1,141 @@
+---
+author: Carlos Zamora @cazamor
+created on: 2019-08-30
+last updated: 2019-09-21
+issue id: <github issue id>
+---
+
+# Keyboard Selection
+
+## Abstract
+
+This spec describes a new set of keybindings that allows the user to create and update a selection without the use of a mouse or stylus.
+
+## Inspiration
+
+ConHost allows the user to edit an existing selection using the keyboard. Holding `Shift` allows the user to move the second selection anchor in accordance with the arrow keys. The selection anchor updates by one cell per key event, allowing the user to refine the selected region. However, mouse selection seems to be superior in that it provides access to the following special selection modes:
+- Double Click: word selection
+- Triple Click: line selection
+
+Mark mode allows the users to update the selection anchors sequentially, and create a block selection.
+
+
+## Solution Design
+
+The fundamental solution design for keyboard selection is that the responsibilities between the Terminal Control and Terminal Core must be very distinct. The Terminal Control is responsible for handling user interaction and directing the Terminal Core to update the selection. The Terminal Core will need to update the selection according to the preferences of the Terminal Control.
+
+### Fundamental Terminal Control Changes
+
+`TermControl::_KeyDownHandler()` is responsible for interpretting the key events. At the time of writing this spec, there are 3 cases handled in this order:
+- Alt+Gr events
+- Keybindings
+- Send Key Event (also clears selection)
+
+Conditionally effective keybindings can be introduced to update the selection accordingly. A well known conditionally effective keybinding is the `Copy` action where text is copied if a selection is active, otherwise the key event is marked as `handled = false`. See "UI/UX Design" --> "Keybindings" for more information regarding keybindings.
+
+### Fundamental Terminal Core Changes
+
+The Terminal Core will need to expose a `KeyboardSelection()` function that is called by the keybinding handler. The following parameters will need to be passed in:

```

Comment:
How does this play with `copyOnSelect`?

Reply:
Added a note. It _should_ update the clipboard on every selection change... Glad we're fixing the crash on copy bug or else this might have resulted in a nightmare.